esp/services/ws_ecl/ws_ecl_wuinfo.cpp

```diff
@@ -209,11 +209,39 @@ bool WsEclWuInfo::getWsResource(const char *name, StringBuffer &out)
         IArrayOf<IPropertyTree> parts;
         if (fields.length())
         {
+            bool addedAdditionalFields = false;
+            unsigned offset = 1;
             ForEachItemIn(i, fields)
             {
-                Owned<IConstWUResult> var = wu->getVariableByName(fields.item(i));
-                if (var)
-                    appendVariableParmInfo(parts, resultSetFactory, *var, i+1);
+                const char *name = fields.item(i);
+                if (!addedAdditionalFields && streq(name, "*")) //name of '*' means insert all unspecified fields here, can only be used once
+                {
+                    addedAdditionalFields=true;
+                    Owned<IConstWUResultIterator> vars = &ensureWorkUnit()->getVariables();
+                    ForEach(*vars)
+                    {
+                        SCMStringBuffer s;
+                        IConstWUResult &var = vars->query();
+                        if (NotFound == fields.find(var.getResultName(s).str()))
+                        {
+                            SCMStringBuffer fieldSeq;
+                            var.getResultFieldOpt("sequence", fieldSeq);
+                            int seq = fieldSeq.length() ? atoi(fieldSeq.str()) : 0;
+                            if (seq < 0)
+                                seq = 0;
+                            seq = seq + i + 1;
```

Comment:
I do not understand the logic here. Does the 'fieldSeq' give the offset from current location i? Can we assume that it returns continuous '1, 2, ..., n'?